Framework/Algorithms/src/MaxMin.cpp

```diff
@@ -89,9 +94,14 @@ void MaxMin::exec() {
   }
 
   // Create the 1D workspace for the output
-  MatrixWorkspace_sptr outputWorkspace =
-      API::WorkspaceFactory::Instance().create(localworkspace,
-                                               MaxSpec - MinSpec + 1, 2, 1);
+  MatrixWorkspace_sptr outputWorkspace;
+  if (localworkspace->id() == "EventWorkspace") {
+    outputWorkspace = create<DataObjects::Workspace2D>(
```

Comment:
See above.

Reply:
Sorry, by "see above" I was refring to the special case handling for "EventWorkspace", not the `auto`.

Reply:
ah, that makes much more sense :)